### PR TITLE
sync typeshed

### PR DESCRIPTION
**Pull request by herb**

### CODE REVIEW

File: mypy/test/data/python2eval.test [hunk 1]
```diff
@@ -399,25 +399,25 @@ print b.x + 1
...
-re.subn(bre, lambda m: b'', b'')[0] + b''
-re.subn(bpat, lambda m: b'', b'')[0] + b''
-[out]
-_program.py:9: error: Cannot infer type argument 1 of "search"
+if False:
+    bre = b'a+'
+    bpat = re.compile(bre)
+    bpat = re.compile(bpat)
+    re.search(bre, b'').groups()
+    re.search(bre, u'') # This ought to be an error, but isn't because of bytes->unicode equivalence
```
Comment (JukkaL) [inline mypy/test/data/python2eval.test]: This comment doesn't make sense

File: mypy/test/data/python2eval.test [hunk 2]
```diff
@@ -399,25 +399,25 @@ print b.x + 1
...
+    re.search(bpat, b'').groups()
+    re.search(bpat, u'') # Error
```
Comment (JukkaL) [inline mypy/test/data/python2eval.test]: This comment is out of date as well

File: mypy/test/data/python2eval.test (hunk 1 above)
Comment (herb) [inline mypy/test/data/python2eval.test]: i don't have context here. should the comment be removed outright?

File: mypy/test/data/python2eval.test (hunk 2 above)
Comment (herb) [inline mypy/test/data/python2eval.test]: removed

**Pull request merged**